misc.go

```diff
@@ -338,6 +346,42 @@ func max(a, b int) int {
 	return b
 }
 
+// This function parses given locale string into language tag value. Passing empty
+// string as locale means reading locale value from environment.
+func getLocaleTag(localeStr string) (language.Tag, error) {
+	localeTag := language.Und
+	var err error
+
+	if localeStr == localeStrSys {
+		// read environment locale
+		return locale.Detect()
+	}
+
+	localeTag, err = language.Parse(localeStr)
+	if err != nil {
+		return localeTag, fmt.Errorf("invalid locale %q: %s", localeStr, err)
+	}
+
+	return localeTag, nil
+}
+
+// This function creates new collator for given locale. Passing empty string as
+// as locale means reading locale value from environment.
+func makeCollator(localeStr string, o ...collate.Option) (*collate.Collator, error) {
+	if localeStr == localeStrDisable {
+		return nil, fmt.Errorf("locale suppose to be disabled with given string")
```

Comment:
The below is shorter and probably just as fine.

```suggestion
		return nil, fmt.Errorf("locale is disabled")
```